core/src/test/java/cz/xtf/core/image/ImageTest.java

```diff
@@ -0,0 +1,102 @@
+
+    @SystemStub
+    private SystemProperties systemProperties;
+
+    @Test
+    public void testResolveImageWithSubId() {
+        systemProperties.set("xtf.eap.image", EAP_IMAGE_URL);
+        systemProperties.set("xtf.eap.subid", EAP_IMAGE_SUBID_NAME);
+        systemProperties.set("xtf.eap.wildfly-openjdk17.image", EAP_IMAGE_SUBID_URL);
+        XTFConfig.loadConfig();
+        Image image = Image.resolve("eap");
+        Assertions.assertNotNull(image, "No image was found.");
+        Assertions.assertEquals(EAP_IMAGE_SUBID_URL, image.getUrl(), "Resolved image is wrong.");
+
+    }
+
+    @Test
+    public void testResolveImageWithoutSubidImage() {
+        systemProperties.set("xtf.eap.image", EAP_IMAGE_URL);
+        systemProperties.set("xtf.eap.subid", EAP_IMAGE_SUBID_NAME);
+        XTFConfig.loadConfig();
+        Image image = Image.resolve("eap");
+        Assertions.assertNotNull(image, "No image was found.");
+        Assertions.assertEquals(EAP_IMAGE_URL, image.getUrl(), "Resolved image is wrong.");
+    }
+
+    @Test
+    public void testResolveImageWithoutSubidName() {
+        systemProperties.set("xtf.eap.image", EAP_IMAGE_URL);
+        systemProperties.set("xtf.eap.wildfly-openjdk17.image", EAP_IMAGE_SUBID_URL);
+        XTFConfig.loadConfig();
+        Image image = Image.resolve("eap");
+        Assertions.assertNotNull(image, "No image was found.");
+        Assertions.assertEquals(EAP_IMAGE_URL, image.getUrl(), "Resolved image is wrong.");
+    }
+
+    @Test
+    public void testResolveWithCustomRegistry() {
+        systemProperties.set("xtf.eap.image", EAP_IMAGE_URL);
+        systemProperties.set("xtf.eap.subid", EAP_IMAGE_SUBID_NAME);
+        systemProperties.set("xtf.eap.wildfly-openjdk17.image", EAP_IMAGE_SUBID_URL);
+        systemProperties.set("xtf.eap.reg", "custom.registry.io");
+        XTFConfig.loadConfig();
+        Image image = Image.resolve("eap");
+        Assertions.assertEquals("custom.registry.io", image.getRegistry());
+        Assertions.assertEquals("wildfly", image.getUser());
+        Assertions.assertEquals("wildfly-s2i", image.getRepo());
+        Assertions.assertEquals("latest-jdk17", image.getTag());
+    }
+
+    @Test
+    public void resolveWithCustomUserTest() {
+        systemProperties.set("xtf.eap.image", EAP_IMAGE_URL);
+        systemProperties.set("xtf.eap.subid", EAP_IMAGE_SUBID_NAME);
+        systemProperties.set("xtf.eap.wildfly-openjdk17.image", EAP_IMAGE_SUBID_URL);
+        systemProperties.set("xtf.eap.user", "customuser");
+        XTFConfig.loadConfig();
+        Image image = Image.resolve("eap");
+        Assertions.assertEquals("quay.io", image.getRegistry());
+        Assertions.assertEquals("customuser", image.getUser());
+        Assertions.assertEquals("wildfly-s2i", image.getRepo());
+        Assertions.assertEquals("latest-jdk17", image.getTag());
+    }
+
+    @Test
+    public void resolveWithCustomTagTest() {
```

Comment:
For completness, I am missing custom repo variant.

Reply:
Looking at `resolve` method, custom repo cannot be set.

Reply:
I see `resolveWithCustomRepoTest()` was added but seems to me same content as `resolveWithCustomTagTest()`. So will be `resolveWithCustomRepoTest()` added or not?

Reply:
Sorry, my fault. I've wanted to add it and then figured out that it's not possible committing with other changes. Now it should be ok.